walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -19,36 +19,58 @@
 import java.util.Collections;
 import java.util.List;
 
-/** Finds possible meeting times for a request for all mandatory participants */
+/** Finds possible meeting times for a request for all mandatory participants and potentially optional participants*/
 public final class FindMeetingQuery {
   /**
    * @param events List of all known events with times and participants
    * @param request Meeting to be planned
    * @return List of all possible meeting times
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<TimeRange> busyTimes = getBusyTimes(events, request);
-    Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
-    busyTimes = mergeIntervals(busyTimes);
-    return getAvailableTimes(busyTimes, request.getDuration());
+    List<TimeRange> mandatoryBusyTimes = getBusyTimes(events, request, "mandatory", Collections.emptyList());
+    List<TimeRange> mandatoryResult = getAvailableTimes(mandatoryBusyTimes, request.getDuration());
+
+    // If there are optional attendees, account for them in schedule
+    if (!request.getOptionalAttendees().isEmpty()) {
```

Comment:
Return early here.

Copied from best practice:
Reduce Code Complexity by Reducing Nesting
Bad: 
```
response = stub.Call(request, rpc)
  if rpc.status == pywraprpc.RPC.OK:
    if response.GetAuthorizedUser():
      if response.GetEnc() == 'utf-8':
        if response.GetRows():
          vals = [ParseRow(r) for r in 
                  response.GetRows()]
          avg = sum(vals) / len(vals)
          return avg, vals
        else:
          raise ValueError('no rows')
      else:
        raise AuthError('unauthorized')
    else:
      raise ValueError('wrong encoding')
  else:
    raise RpcError(rpc.ErrorText())

```
Good:
```
response = stub.Call(request, rpc)

  if rpc.status != pywraprpc.RPC.OK:
    raise RpcError(rpc.ErrorText())


  if not response.GetAuthorizedUser():
    raise ValueError('wrong encoding')

  if response.GetEnc() != 'utf-8':
    raise AuthError('unauthorized')

  if not response.GetRows():
    raise ValueError('no rows')

  vals = [ParseRow(r) for r in 
          response.GetRows()]
  avg = sum(vals) / len(vals)
  return avg, vals

```

Reply:
done, thank you!